src/lib/converter/ts-internal.ts

```diff
@@ -0,0 +1,130 @@
+import * as ts from "typescript";
+
+/**
+ * Expose the internal TypeScript APIs that are used by TypeDoc
+ */
```

Comment:
> any concern about the out-datedness that could occur

That is certainly a concern.  My hope is that these internal usages can be replaced by other APIs.